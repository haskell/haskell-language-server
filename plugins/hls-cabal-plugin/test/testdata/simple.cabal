cabal-version:      3.0
name:               hls-cabal-plugin
version:            0.1.0.0
synopsis:
homepage:
license:            MIT
license-file:       LICENSE
author:             Fendor
maintainer:         fendor@posteo.de
category:           Development
extra-source-files: CHANGELOG.md

library
    exposed-modules:  IDE.Plugin.Cabal
    build-depends:    base ^>=4.14.3.0
    hs-source-dirs:   src
    default-language: Haskell2010

test-suite hls-cabal-plugin-test
    default-language: Haskell2010
    type:             exitcode-stdio-1.0
    hs-source-dirs:   test
    main-is:          Main.hs
    build-depends:    base ^>=4.14.3.0
