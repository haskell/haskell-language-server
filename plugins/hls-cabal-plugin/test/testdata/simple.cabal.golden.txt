([],Right (GenericPackageDescription {packageDescription = PackageDescription {specVersion = CabalSpecV3_0, package = PackageIdentifier {pkgName = PackageName "hls-cabal-plugin", pkgVersion = mkVersion [0,1,0,0]}, licenseRaw = Left (License (ELicense (ELicenseId MIT) Nothing)), licenseFiles = [SymbolicPath "LICENSE"], copyright = "", maintainer = "fendor@posteo.de", author = "Fendor", stability = "", testedWith = [], homepage = "", pkgUrl = "", bugReports = "", sourceRepos = [], synopsis = "", description = "", category = "Development", customFieldsPD = [], buildTypeRaw = Nothing, setupBuildInfo = Nothing, library = Nothing, subLibraries = [], executables = [], foreignLibs = [], testSuites = [], benchmarks = [], dataFiles = [], dataDir = ".", extraSrcFiles = ["CHANGELOG.md"], extraTmpFiles = [], extraDocFiles = []}, gpdScannedVersion = Nothing, genPackageFlags = [], condLibrary = Just (CondNode {condTreeData = Library {libName = LMainLibName, exposedModules = [ModuleName "IDE.Plugin.Cabal"], reexportedModules = [], signatures = [], libExposed = True, libVisibility = LibraryVisibilityPublic, libBuildInfo = BuildInfo {buildable = True, buildTools = [], buildToolDepends = [], cppOptions = [], asmOptions = [], cmmOptions = [], ccOptions = [], cxxOptions = [], ldOptions = [], hsc2hsOptions = [], pkgconfigDepends = [], frameworks = [], extraFrameworkDirs = [], asmSources = [], cmmSources = [], cSources = [], cxxSources = [], jsSources = [], hsSourceDirs = [SymbolicPath "src"], otherModules = [], virtualModules = [], autogenModules = [], defaultLanguage = Just Haskell2010, otherLanguages = [], defaultExtensions = [], otherExtensions = [], oldExtensions = [], extraLibs = [], extraGHCiLibs = [], extraBundledLibs = [], extraLibFlavours = [], extraDynLibFlavours = [], extraLibDirs = [], includeDirs = [], includes = [], autogenIncludes = [], installIncludes = [], options = PerCompilerFlavor [] [], profOptions = PerCompilerFlavor [] [], sharedOptions = PerCompilerFlavor [] [], staticOptions = PerCompilerFlavor [] [], customFieldsBI = [], targetBuildDepends = [Dependency (PackageName "base") (MajorBoundVersion (mkVersion [4,14,3,0])) (fromNonEmpty (LMainLibName :| []))], mixins = []}}, condTreeConstraints = [Dependency (PackageName "base") (MajorBoundVersion (mkVersion [4,14,3,0])) (fromNonEmpty (LMainLibName :| []))], condTreeComponents = []}), condSubLibraries = [], condForeignLibs = [], condExecutables = [], condTestSuites = [(UnqualComponentName "hls-cabal-plugin-test",CondNode {condTreeData = TestSuite {testName = UnqualComponentName "", testInterface = TestSuiteExeV10 (mkVersion [1,0]) "Main.hs", testBuildInfo = BuildInfo {buildable = True, buildTools = [], buildToolDepends = [], cppOptions = [], asmOptions = [], cmmOptions = [], ccOptions = [], cxxOptions = [], ldOptions = [], hsc2hsOptions = [], pkgconfigDepends = [], frameworks = [], extraFrameworkDirs = [], asmSources = [], cmmSources = [], cSources = [], cxxSources = [], jsSources = [], hsSourceDirs = [SymbolicPath "test"], otherModules = [], virtualModules = [], autogenModules = [], defaultLanguage = Just Haskell2010, otherLanguages = [], defaultExtensions = [], otherExtensions = [], oldExtensions = [], extraLibs = [], extraGHCiLibs = [], extraBundledLibs = [], extraLibFlavours = [], extraDynLibFlavours = [], extraLibDirs = [], includeDirs = [], includes = [], autogenIncludes = [], installIncludes = [], options = PerCompilerFlavor [] [], profOptions = PerCompilerFlavor [] [], sharedOptions = PerCompilerFlavor [] [], staticOptions = PerCompilerFlavor [] [], customFieldsBI = [], targetBuildDepends = [Dependency (PackageName "base") (MajorBoundVersion (mkVersion [4,14,3,0])) (fromNonEmpty (LMainLibName :| []))], mixins = []}}, condTreeConstraints = [Dependency (PackageName "base") (MajorBoundVersion (mkVersion [4,14,3,0])) (fromNonEmpty (LMainLibName :| []))], condTreeComponents = []})], condBenchmarks = []}))